### ISSUE TITLE
Ideas to improve Model

### ISSUE THREAD

**Issue by blumenstiel** [blumenstiel/TDT4265-project#4]

**Comment by marianasonja** [comment]
Ideas to improve model (used throughout A1-A4) 

**Network Parameters:** 

- [ ] Change  Learning Rate

  Initial learning rate: 

- [ ] Use different optimizers: Adam Optimizer vs. SGD vs. Momentum 

  Adam Optimizer so the network converges much faster due to its adaptive learning rate

- [ ]  Leaky ReLu: 

  Use leaky ReLu function because Leaky ReLU copes with the "dead neuron" problem of ReLu 
 (gradient is 0 for all negative values and no learning is performed in the layers below). 

- [ ]  Batch Normalization: 

  Use Batch Normalization to stabilize and speed up the training and make the initial weights less relevant

- [ ]  Change Output Channels 

- [ ]  Change filter size/kernel size 

- [ ]  Change Stride 

- [ ] Try MaxPooling2D vs Average Pooling 

- [ ]  Use different/mutliple shapes for anchor boxes

- [ ] Use pretrained model 

**Data Parameters**

- Data augmentation: [Data Augmentation](https://github.com/blumenstiel/TDT4265-project/issues/3)

**Comment by blumenstiel** [comment]
- [ ] Customize PIXEL_MEAN, PIXEL_STD

**Comment by blumenstiel** [comment]
- [ ] Compare Adam to Momentum Optimizer

**Comment by blumenstiel** [comment]
- [ ] Dropout

**Comment by blumenstiel** [comment]
- [ ] Early stopping ?

**Comment by marianasonja** [comment]
- [ ] Batch Normalization 
- [ ] L2 Regularization 
- [ ] Combining Batch Normalization and L2 Regularization --> apparently very powerful 
[Reference](https://blog.janestreet.com/l2-regularization-and-batch-norm/)

**Comment by marianasonja** [comment]
- [ ] Min Pooling

We should try to use Minimum Pooling (instead of Max Pooling) as this would help to identify dark objects on a lighter background.

**Comment by marianasonja** [comment]
- [ ] Try out Grayscale --> Output Dim from 3 to 1
CLASS torchvision.transforms.Grayscale(num_output_channels=1)
